cirq-core/cirq/sim/act_on_density_matrix_args.py

```diff
@@ -33,25 +33,30 @@ class ActOnDensityMatrixArgs(ActOnArgs):
     storing the density matrix of the quantum system with one axis per qubit.
     """
 
+    @_compat.deprecated_parameter(
+        deadline='v0.15',
+        fix='Use initial_state instead.',
```

Comment:
Thinking more about this, maybe the best thing to do rather than introduce a new duplicate parameter is to say "Don't pass by keyword" here. That way initial state is just the first argument to the function call and allows us to change the variable name at will. The way we have it currently, there are still edge cases that could be broken in the future when we move initial_state to the top (assuming we want to do so).

Or I guess alternately we could force passing by name, if we put a `*,` before line 51, and just leave it there.

@95-martin-orion do you have a preference?
@95-martin-orion

Reply:
I *think* what I'd prefer here is to migrate toward forcing everything to be passed by name here. Especially since all the arguments are optional and we can handle different combinations of them, this function seems to be begging for pass-by-name style. So maybe what I'd recommend is deprecating `target_tensor` as we do here, *and* deprecating `len(args) != 0)` to ensure keyword style calling. Then later when the deprecation is complete we can add `*,` at the beginning to enforce it. Haven't tried this with `@deprecated_parameter` but it seems like it should work. @95-martin-orion thoughts?

Reply:
It's a little hacky, but I'd be OK signing off on that application of `@deprecated_parameter`. I assume the other act_on_args would receive similar treatment?

Reply:
Yes, the others would be handled the same way. @yjt98765 you can go ahead with deprecating `len(args) != 0` as a step toward enforcing keyword-style args passing (in addition to your existing logic for deprecating `target_tensor` itself).

Reply:
Sure. I will update them soon.